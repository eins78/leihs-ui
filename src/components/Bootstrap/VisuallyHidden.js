import React from 'react'
import PropTypes from 'prop-types'
// import cx from 'classnames'

const VisuallyHidden = ({ children, ...props }) => {
  const hidden = !!props['if']
  return (
    <div
      className={hidden ? 'sr-only' : ''}
      aria-hidden={hidden}
      tabIndex={hidden ? -1 : undefined}
    >
      {children}
    </div>
  )
}
VisuallyHidden.propTypes = { if: PropTypes.bool.isRequired }

export default VisuallyHidden
