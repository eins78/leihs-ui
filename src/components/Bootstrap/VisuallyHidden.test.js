import React from 'react'
import renderer from 'react-test-renderer'

import VisuallyHidden from './VisuallyHidden'

const examples = [
  {
    content: (
      <form>
        <VisuallyHidden if={false}>
          Inner Text <input type="text" />
        </VisuallyHidden>
      </form>
    )
  },
  {
    content: (
      <form>
        <VisuallyHidden if={true}>
          Inner Text <input type="text" />
        </VisuallyHidden>
      </form>
    )
  }
]

examples.forEach(({ name, content }) => {
  it(`${name}: renders correctly`, () => {
    expect(renderer.create(content).toJSON()).toMatchSnapshot()
  })
})
