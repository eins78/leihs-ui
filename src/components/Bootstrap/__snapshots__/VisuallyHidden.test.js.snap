// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined: renders correctly 1`] = `
<form>
  <div
    aria-hidden={false}
    className=""
  >
    Inner Text 
    <input
      type="text"
    />
  </div>
</form>
`;

exports[`undefined: renders correctly 2`] = `
<form>
  <div
    aria-hidden={true}
    className="sr-only"
    tabIndex={-1}
  >
    Inner Text 
    <input
      type="text"
    />
  </div>
</form>
`;
